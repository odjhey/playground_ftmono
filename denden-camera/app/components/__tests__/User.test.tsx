/**
 * @format
 */

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import UserProfile, { UserItem } from '../components/User';

describe('UserProfile', () => {
  describe('Rendering', () => {
    const userData = {
      name: 'Name',
      checkin: false,
    };
    it('Renders', () => {
      shallow(<UserProfile name={userData.name} checkin={userData.checkin} />);
    });
    it('UserItem Renders', () => {
      expect(
        shallow(<UserItem name={userData.name} checkin={userData.checkin} />),
      ).toMatchSnapshot();
    });
  });

  describe('Interaction', () => {
    const userData = {
      name: 'Name',
      checkin: false,
    };
    it('Profile checkin value should change after Checkin Pressed', () => {
      renderer.create(
        <UserProfile name={userData.name} checkin={userData.checkin} />,
      );
    });
    it('Profile checkin value should change after Checkin Pressed', () => {
      renderer.create(
        <UserProfile name={userData.name} checkin={userData.checkin} />,
      );
    });
  });
});
