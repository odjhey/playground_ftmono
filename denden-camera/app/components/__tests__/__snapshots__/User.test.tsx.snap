// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<S> UserProfile UserItem Renders 1`] = `
<Themed.ListItem
  leftAvatar={
    <Themed.Avatar
      onPress={[Function]}
      rounded={true}
      title="O"
    />
  }
  rightElement={
    <View>
      <Themed.Badge
        status="error"
        value="Online Status"
      />
    </View>
  }
  title="Name"
/>
`;

exports[`UserProfile Rendering UserItem Renders 1`] = `
<Themed.ListItem
  leftAvatar={
    <Themed.Avatar
      onPress={[Function]}
      rounded={true}
      title="O"
    />
  }
  rightElement={
    <View>
      <Themed.Badge
        status="error"
        value="Online Status"
      />
    </View>
  }
  title="Name"
/>
`;
